cypress/integration/MultiSelect/common/index.js

```diff
@@ -42,3 +46,9 @@ Then('the options are displayed', () => {
 Then('the MultiSelect has focus', () => {
     cy.focused().should('have.attr', 'data-test', 'dhis2-uicore-select-input')
 })
+
+When('an option is clicked', () => {
+    cy.contains('option one').click()
+    // Wait until the DOM is painted
+    cy.wait(100)
```

Comment:
This worries me. I could tell from the snapshots I was dealing with a timing issue. Initially I added a timeout of 1 ms, which I thought was justifiable as a poor-man's substitute for `requestAnimationFrame`. This worked fine locally, but after pushing this commit, and upgrading some dependencies, the tests in GitHub actions started failing. I have only been able to resolve these issues this way, but I fear there is an underlying reason that these tests are failing.

Reply:
My worries turned out to be justified. All tests connected to the popper's resize-observer behaviour were flaky. Some test runs would pass and others would fail. There didn't seem to be a pattern in it. I've spent most of today trying to find a decent solution for this, either in the component or in the form of a cypress helper.....

Since the behaviour of the component actually seemed correct when testing manually and when inspecting the DOM snapshots in cypress, I chose to try and fix this on the cypress side. I've come up with a helper function that uses a ResizeObserver itself. Since implementing that I've had a series successful test runs, so hopefully the problem is now resolved....